### DISCUSSION TITLE
Read chat log on the client-side

### DISCUSSION THREAD

**Discussion by MasterBroNetwork** [FabricMC/fabric-api#1851] [Mod Dev Support]
Is it possible to read the chat log on the client-side, get the information of the chat log and if it matches a specific message then run some code?

I've been looking into onChatMessage and ChatMessageC2SPacket and stuff but I'm at a dead end, help with this is massively appreciated, thanks.

**Comment by warjort** [comment]
ChatMessageC2SPacket is C2S (client to server).

For the messages sent to client you want GameMessageS2CPacket (S2C = server to client).

**Comment by warjort** [comment]
Probably the best place to introduce some code is the inGameHud class
That has a 
````
private final Map<MessageType, List<ClientChatListener>> listeners
````
You could use an access widener or mixin accessor to allow you to add your own ClientChatListener for MessageType=CHAT

**Comment by warjort** [comment]
You can get access to the inGameHud instance in your mod's client initializer using
````
InGameHud hud = MinecraftClient.getInstance().inGameHud
````
